### PR TITLE
Adds the ability to define an escalation policy for the manage incidents API

### PR DESCRIPTION
**Pull request by evnsio**

### CODE REVIEW

**Review by theckman — APPROVED** [review]

File: incident.go [hunk 1]
```diff
@@ -154,11 +154,12 @@ type CreateIncidentOptions struct {
...
-	Type        string        `json:"type"`
-	Status      string        `json:"status,omitempty"`
-	Priority    *APIReference `json:"priority,omitempty"`
-	Assignments []Assignee    `json:"assignments,omitempty"`
+	ID               string            `json:"id"`
+	Type             string            `json:"type"`
+	Status           string            `json:"status,omitempty"`
+	Priority         *APIReference     `json:"priority,omitempty"`
+	Assignments      []Assignee        `json:"assignments,omitempty"`
+	EscalationPolicy *EscalationPolicy `json:"escalation_policy,omitempty"`
```
Comment (theckman) [inline incident.go:162]: While I know this is technically an `EscalationPolicy`, looking at the [PagerDuty API documentation](https://developer.pagerduty.com/api-reference/reference/REST/openapiv3.json/paths/~1incidents/put) it seems like this expects the [APIObject](https://pkg.go.dev/github.com/PagerDuty/go-pagerduty#APIObject) subset of the `EscalationPolicy` type.

I'm not sure if the PagerDuty will always return success if we rendered the other `EscalationPolicy` fields in the JSON document. That makes me think we should update this to be an `*APIObject`.

What are your thoughts?

**Review by theckman — CHANGES_REQUESTED** [review]

**Comment by theckman** [comment]
@evnsio I wanted to see if you saw my comment above, and whether you might have the spare cycles to make the suggested iteration?

**Comment by theckman** [comment]
@evnsio I wanted to ping again to see if you could make the `*APIObject` change above. This is looking like it'll soon be the only thing left open for `v.1.4.0`.

**Comment by evnsio** [comment]
Sorry I completely missed these notifications.  I'll take a look over the weekend if that works?

**Comment by theckman** [comment]
@evnsio absolutely! I really appreciate you making the time.

**Comment by theckman** [comment]
@evnsio just wanted to ping a final time. This is the final item outstanding for `v1.4.0`, and I'd like to get us towards releasing that. If you're not free it's no sweat, please let me know and I'm happy to make this change.

**Comment by theckman** [comment]
Thank you for raising this PR / feature request. I've raised a complimentary PR (#323) that supersedes this one. I'll close this PR at this time.

Although we did not get to merge your PR, we appreciate the contribution!

**Pull request closed without merging**